musings/amd-sev.md

```diff
@@ -0,0 +1,47 @@
+# AMD Secure Encrypted Virtualization (AMD-SEV)
+
+# Concepts
+- AMD-SEV is a targeted at securing virtual machines by encrypting the memory of each virtual machine with a unique key.
+- SEV can protect your machine from a portentially malicious hypervisor.
+- SEV can calculate a signature of virtual machine's memory content, which can be sent to the VM's owner as an attestation that the memory on the target host, was encrypted correctly by firmware.
+- AMD-SEV SNP is and extension to SEV which adds new hardware-based security protections
+
+## Keys used in SEV
+The AMD SEV firmware provides a mechanism to verify that it is executing on AMD hardware that supports SEV. Various keys are used in this process:
+
+- $PDH$ (Platform Diffie Hellman) key - This key is used to negotiate a master secret which is then used with a key derivation function to establish a trusted channel
+- $PEK$ (Platform Endorsement Key) - This key signs the $PDH$ to anchor the $PDH$ to the AMD root of trust and the platform owner's root of trust
+- $CEK$ (Chip endorsement key) - This key signs the $PEK$ to anchor the $PEK$ to the AMD root of trust. Each chip has a unique $CEK$ which is derived from secrets stored in the chip's one-time programmable (OTP) memory
+- $ASK$ (AMD Signing Key) - The $ASK$ private key signs the $CEK$ public key  to demonstrate that the $CEK$ is an authentic AMD key
+- $ARK$ (AMD Root Key) - The $ARK$ private key signs the $ASK$ public key to demonstrate that the $ASK$ is an authentic AMD key. This key is the root of trust of AMD and its signatures signify AMD authencity
+
+Therefore the following certificate chain is produced:
+
+```
+ARK -> ASK -> CEK -> PEK -> PDH
+```
+
+Therefore if the secure channel can be established using the $PDH$ key, then it is ensured that this is an authentic AMD system which has the SEV feature.
+
+## AMD-SEV SNP Attestation report measurements
+
+### Platform measurements
+- CHIP_ID - The unique chip identifier
+- PLATFORM_INFO - Indicates properties of the platform configuration, for example whether whole system memory encryption (TSME) or simultaneous multithreading (SMT) is enabled
+- CURRENT_TCB - Security Version Numbers (SVNs) of the current executing platform firmware and microcode
+- COMMITTED_TCB - SVNs of the anti-rollback minimum of the platform firmware and microcode
+- REPORTED_TCB - SVN of the hypervisor. The hypervisor has the option to report a lower version
+- LAUNCH_TCB - SVNs of the platform firmware and microcode at the time the guest was launched or imported
```

Comment:
I agree, I was also deliberating on this aspect of who should verify these fields. My thoughts: yes both guest and platform measurements could be verified by Veraison since the guest is inevitably tied to the platform.

The *_TCB fields are SVNs which can potentially be captured in a the SVN field within the comid Mval struct. Platform_info and chip ID could be within the within RawValue since there arent other more appropriate fields within Mval that fit this.